pkg/cloudprovider/aws/packing/packing.go

```diff
@@ -78,6 +80,42 @@ func (p *podPacker) Pack(ctx context.Context, pods []*v1.Pod) ([]*Packing, error
 
 // TODO filter instance types based on node contraints like availability zones etc.
 func (p *podPacker) getNodeCapacities() []*nodeCapacity {
+	nodeCapacities := []*nodeCapacity{}
+	describeInstanceTypesInput := &ec2.DescribeInstanceTypesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("processor-info.supported-architecture"),
+				Values: []*string{aws.String("x86_64")},
+			},
+			{
+				Name:   aws.String("supported-usage-class"),
+				Values: []*string{aws.String("on-demand")},
+			},
+			{
+				Name:   aws.String("supported-virtualization-type"),
+				Values: []*string{aws.String("hvm")},
+			},
+		},
+	}
+	err := p.ec2.DescribeInstanceTypesPagesWithContext(context.TODO(), describeInstanceTypesInput, func(page *ec2.DescribeInstanceTypesOutput, lastPage bool) bool {
```

Comment:
We should cache this result (similar to launch template cache), as the data changes very rarely and we don't want to spam the API.

Reply:
I haven't look at the launch template cache yet. I'll probably do a follow up PR to add that, if it's okay with you?

Reply:
Sounds good - can you add an issue?

Reply:
We might want to consider an instanceTypeProvider struct that follows the exact code patterns as Launch Template, Subnet, Security Group, etc.

Reply:
Here's the issue for adding the cache: https://github.com/awslabs/karpenter/issues/292